
---
---
<br/>

# **Discerning Ancestry-Based Assortative Mating from Migration by their Genomic Imprints upon Admixed Populations** 



<br/><br/><br/>

# Contents

<br/>

 * Description

 * Files list
  
 * Requirements
 
 * Contributions
 
 * Credits

<br/>


<br/><br/><br/>

# Description
<br/>


This is my final research project of the CMEE course at Silwood Campus, Imperial College London. <br/><br/>

All work was done through VSCode on a system running Ubuntu 20.04 and/or Imperial College London's HPC cluster. <br/><br/>

The project comprises a workflow, in the form of a series of scripts. It integrates genomic data with slave migration data to investigate historic assortative mating in six admixed populations of the Americas: Barbados, Colombia, Mexico, Peru, Puerto Rico and the US Southwest. Please see the file Ben_Nouhan_Report.pdf in the ../writeup/ dirrectory for further understanding of the background behind this project, and the structure of the workflow.
  
<br/>

**Author: &nbsp; Ben Nouhan, &nbsp; bjn20@ic.ac.uk**
<br/><br/>


<br/><br/><br/>

# File List
<br/>


admixture_analysis.R &emsp;-&emsp; Makes stacked barcharts of Admixture output and save dataframe used as .rda file

make_query_list.py &emsp;-&emsp; Generates an RFMix2-compatible query sample map to communicate for which samples we want local ancestry inference

make_ref_samplemap.py &emsp;-&emsp; Uses ADMIXTURE output to filter out samples without more than 99% ancestry of one of the ancestries, so those remaining can be used as RFMix2 reference samples

msp2bed_conversion.R &emsp;-&emsp; Converts .msp.tsv RFMix2 output files into .bed files for use by TRACTS

rfmix.Q_analysis.R &emsp;-&emsp; Makes stacked barcharts of RFMix2 output and save dataframe used as .rda file

rfmix.fb.tsv_genotype_analysis.R &emsp;-&emsp; Analyses genotype assignments made by rfmix.fb.tsv_genotype_assign_HPC.R and use this to calculate biallelic and triallelic assortative mating index values for each ancestry-informative marker position, before using them for various plots

rfmix.fb.tsv_genotype_assign_HPC.R &emsp;-&emsp; Analyses RFMix2 output .fb.tsv file, using haplotypes to assign genotypes for more efficiet analysis later on which can be done on the local machine

rfmix.msp.tsv_inbred_window_size.R &emsp;-&emsp; Analyses RFMix output to generate ancestry-specific distributions of homozygous continuous ancestry tract lengths, and to make histograms based on said distributions

rfmix.msp.tsv_window_size.R &emsp;-&emsp; Analyses RFMix output to generate ancestry-specific distributions of haplotype continuous ancestry tract lengths, and to make histograms, box plots and p-value heatmaps based on said distributions

run_admixture.sh\* &emsp;-&emsp; Runs ancestry proportion estimation using ADMIXTURE on the phased genomic data

run_fb_assignment.sh\* &emsp;-&emsp; Runs rfmix.fb.tsv_genotype_assign_HPC.R on each chromosome in parallel on a HPC

run_rfmix2.sh\* &emsp;-&emsp; Runs local ancestry inference with RFMix2 on the query samples using samples selected by make_ref_samplemap.py as reference

run_shapeit4.sh\* &emsp;-&emsp; Runs phasing with SHAPEIT4 on the genomic data

scatterplot_rfmix_vs_admix.R &emsp;-&emsp; Makes scatterplot comparing outputs of RFMix2 and ADMIXTURE

tracts_mig_plots.R &emsp;-&emsp; Makes plots of ancestry proportion over time fromm tracts output, and merge them with slave voyage data



                  
<br/><br/>

## **tracts_py2/** 

<br/>

models_4pop.py\*\* &emsp;-&emsp; Contains the functions needed by the 4 population model

run_tracts.sh\*\* &emsp;-&emsp; Runs the TRACTS analysis for this project for each population, using .bed files generated by msp2bed_conversion.R

taino_ppxx_xxpp.py\*\* &emsp;-&emsp; Runs the 4 population model on the argued population

tracts.py\*\* &emsp;-&emsp; Module containing various functions and definitions for using other TRACTS scripts

<br/><br/>


\* scripts that must be run on the Imperial HPC cluster rather than a local machine

<br/>

\*\* scripts with the .py extension that use Python 2 rather than Python 3




<br/><br/>


# Requirements
<br/>


## **Software:**
<br/>

BCFtools v1.9 <br/>
SHAPEIT v4.2.0 <br/>
PLINK v2.0 <br/>
ADMIXTURE v1.3.0 <br/>
RFMIX v2.03-r0 <br/>
TRACTS


<br/>



## **Languages:**
<br/>

[![](https://img.shields.io/badge/Python-3.9.0-blue.svg)](https://www.python.org/downloads/release/python-390/) &emsp;
[![](https://img.shields.io/badge/Python-2.7.18-green)](https://www.python.org/downloads/release/python-2718/) &emsp;
[![](https://img.shields.io/badge/R-4.0.3-red)](https://cran.r-project.org/) &emsp;
![](https://img.shields.io/badge/Bash-5.0-white)

<br/>

## **Libraries:**
<br/>

### Python 2
<br/>

[![](https://img.shields.io/badge/numpy--black)](https://numpy.org/install/) &emsp;
[![](https://img.shields.io/badge/scipy--orange)](https://www.scipy.org/install.html) &emsp;
[![](https://img.shields.io/badge/matplotlib-2.0.2-yellow)](https://matplotlib.org/2.0.2/users/installing.html) &emsp;
[![](https://img.shields.io/badge/brewer2mpl-1.4.1-purple)](https://pypi.org/project/pandas/)



### R
<br/>

[![](https://img.shields.io/badge/tidyverse-1.3.0-teal.svg)](https://cran.r-project.org/web/packages/tidyverse/index.html) &emsp;
[![](https://img.shields.io/badge/cowplot-1.1.1-blue)](https://cran.r-project.org/web/packages/cowplot/readme/README.html) &emsp;
[![](https://img.shields.io/badge/qwraps2-0.5.2-orange)](https://cran.r-project.org/web/packages/qwraps2/index.html) &emsp;
[![](https://img.shields.io/badge/qpcR-1.4.1-brown)](https://cran.r-project.org/web/packages/qpcR/index.html) &emsp;
[![](https://img.shields.io/badge/data.table-1.14.0-green)](https://cran.r-project.org/web/packages/data.table/index.html) 



<br/><br/><br/><br/>

All packages not mentioned herein are part of the default installations of Python v3.9.0, Python v2.7.18 or R v4.0.3.

<br/><br/>




# Contributions
<br/>

I am not currently looking for contributions, but feel free to send me any suggestions related to the project at &ensp; b.nouhan.20@imperial.ac.uk

<br/>


<br/><br/><br/>

# Credits
<br/>

This project was carried out as part of the Fumagalli Lab, Faculty of Natural Sciences, Department of Life Sciences (Silwood Park), Imperial College London.<br/>
Special thanks to  Dr Matteo Fumagalli and Dr Alex Mas Sandoval for their help and guidance.

<br/><br/><br/><br/><br/>

---
---