
---
---
<br/>

# **CMEE Coursework Week 2: Python 1** 



<br/><br/><br/>

# Contents

<br/>

 * Description

 * Files list
  
 * Requirements
 
 * Contributions
 
 * Credits

<br/>


<br/><br/><br/>

# Description
<br/>


This is all the assessed coursework from Week 2 (12th Oct - 18th Oct 2020) of the CMEE course at Silwood Campus, Imperial College London. <br/><br/>

All work was done through VSCode on a system running Ubuntu 20.04, and is explained in further details in The Multilingual Quantitative Biologist book (link below).<br/><br/>


Topics covered this week include introductions to:

 - Basics of Python syntax and data structures

 - Pythonâ€™s object-oriented features

 - Learning to use the ipython environment

 - How to write and run python code

 - Understand and implement Python control flow tools

 - Writing, debugging, using, and testing Python functions

<br/><br/>

  
<br/>

**Author: &nbsp; Ben Nouhan, &nbsp; bjn20@ic.ac.uk**
<br/><br/>


<br/><br/><br/>

# File List
<br/>

## **week2/code/**
<br/>
align_seqs.py &emsp;-&emsp; Aligns two default sequences, gives optimum alignment and its number of matching bases

align_seqs_better.py &emsp;-&emsp; Aligns two default or argued FASTA sequences, gives optimum alignments and their number of matching bases

align_seqs_fasta.py &emsp;-&emsp; Aligns two default or argued FASTA sequences, gives optimum alignment and its number of matching bases

basic_csv.py &emsp;-&emsp; Reads .csv file (data/testcsv.csv), extracts species name and bodymass columns, and writes into a new file

basic_io1.py &emsp;-&emsp; Opens a file for subsequent reading

basic_io2.py &emsp;-&emsp; Save the elements of a list to a file

basic_io3.py &emsp;-&emsp; Pickle dumps an object for later use

boilerplate.py &emsp;-&emsp; Prints off a simple message to help explain programmes and modules

cfexercises1.py &emsp;-&emsp; Conditional functions for various calculations

cfexercises2.py &emsp;-&emsp; Examples of loops containing conditionals

control_flow.py &emsp;-&emsp; Some functions exemplifying the use of control statements

debugme.py &emsp;-&emsp; Bugged script by design, for debugging console practice

dictionary.py &emsp;-&emsp; Script to populate a dictionary called taxa_dic from a list of taxa

lc1.py &emsp;-&emsp; Scripts to extract data from a list of tuples into lists

lc2.py &emsp;-&emsp; Scripts to extract specific tuples from a list of tuples

loops.py &emsp;-&emsp; Examples of for and while loops

oaks.py &emsp;-&emsp; Scripts to extract and manipulate data from a list of tree taxa

oaks_debugme.py &emsp;-&emsp; Searches input csv file for species of oak tree and writes them into a new csv document

scope.py &emsp;-&emsp; Collection of scripts demonstrating and explaining variable scope

sysargv.py &emsp;-&emsp; Scripts to demonstrate the use of (sys.argv)

test_control_flow.py &emsp;-&emsp; Copy of control_flow.py, used to demonstrate the doctest module

tuple.py &emsp;-&emsp; Script to print objects within a tuple of tuples, in two ways

using_name.py &emsp;-&emsp; Script to demonstrate python programmes being used as such, or as a module
                  
<br/><br/>

## **week2/data/** 

<br/>

407228326.fasta &emsp;-&emsp; FASTA file used for sequence alignments by align_seqs programmes

407228412.fasta &emsp;-&emsp; FASTA file used for sequence alignments by align_seqs programmes

E.coli.fasta &emsp;-&emsp; FASTA file used for sequence alignments by align_seqs programmes

TestOaksData.csv &emsp;-&emsp; List of tree taxa used by oaks.py and oaks_debugme.py

seq.csv &emsp;-&emsp; Two short sequences used for sequence alignment by align_seqs.py

test.txt &emsp;-&emsp; Lines of text used for a demonstration by basic_io1.py

testcsv.csv &emsp;-&emsp; Table in CSV format giving details of some species, used by basic_csv.py

 

 
<br/>


<br/><br/><br/>

# Requirements
<br/>

## **Languages:**
<br/>

[![](https://img.shields.io/badge/Python-3.9.0-blue.svg)](https://www.python.org/downloads/release/python-390/) &emsp;
![](https://img.shields.io/badge/Bash-5.0-red) &emsp;


<br/>

## **Libraries:**
<br/>

#### Python

[![](https://img.shields.io/badge/numpy-1.18.1-purple)](https://pypi.org/project/numpy/) &emsp;
[![](https://img.shields.io/badge/ipdb-0.13.4-green)](https://pypi.org/project/ipdb/) &emsp;



<br/>


<br/><br/><br/>

# Contributions
<br/>


I am not currently looking for contributions, but feel free to send me any suggestions related to the project at &ensp; b.nouhan.20@imperial.ac.uk

<br/>


<br/><br/><br/>

# Credits
<br/>

This project was (almost exclusively) inspired by The Multilingual Quantitative Biologist book: (https://mhasoba.github.io/TheMulQuaBio/intro.html)<br/>
Special thanks to Dr Samraat Pawar, Pok Ho and Francis Windram for their help.

<br/><br/><br/><br/><br/>

---
---