\documentclass[11pt]{article}

\usepackage[justification=centering]{caption}
\usepackage[a4paper, total={6.5in, 9in}]{geometry}
\emergencystretch=1em
\usepackage{titling}
\setlength{\droptitle}{8em}
\usepackage{xurl}
\usepackage{lineno}
\usepackage[skip=0pt,font=scriptsize]{caption} 
\usepackage[labelfont=bf, justification=justified]{caption}
\usepackage{graphicx}
\usepackage{pgfplotstable,booktabs}
\usepackage{array,ragged2e}
\usepackage{multirow}
\pgfplotsset{compat=1.16}
\usepackage{setspace}
\usepackage[sorting=nyt,style=apa]{biblatex}
\usepackage{booktabs}

\bibliography{library.bib}

% titles:
% Discerning Ancestry-Based Assortative Mating from Migration by their Genomic Imprint upon Admixed Populations of the Americas
\title{Discerning Ancestry-Based Assortative Mating from Migration by their Genomic Imprint upon Admixed Populations of the Americas}
\author{\\ \\ \\ \\ Ben Nouhan, bjn20@ic.ac.uk \\ \\ Imperial College London \\}
\date{\today}
\newcommand\wordcount{\input{report.sum}}


% general instructions
% • Structure and Style:
% All reports should have an Abstract, Introduction, Methods, Results, Discussion and possibly, a separate Conclusions section. Do not format your report to look like a paper from a specific journal.
% • Word limit:
% The main text should not exceed 6,000 words in length (excluding figures, tables, references and appendices/supplementary information). Using the full 6,000 words is often not advisable – this is an absolute maximum. Clarity and brevity are better than wordiness, so do not puff up your thesis to reach 6,000 words. More instructions on content are provided below. You must not abuse the exclusion of figure and table legends from the word count to squeeze in more material. They must only explain the contents of the figure or table!
% • Figure Limit:
% Most journals have an upper limit of the total number of figures and tables – it is rare to see more than a total of about 6 – 8 figures and tables (not 6 – 8 of each!). We would like you to try and keep under a total of 8 figures and tables. If you have more than this, then think carefully about whether they are all crucial to helping the reader understand your research. If they are, then include them; if not, then move them to supplementary material. Published papers often use complex multipart figures to reduce figure counts. Bringing related figures and tables together is good practice, but is often very time consuming and fiddly. You should prioritize making your selected figures and tables as clear and informative as possible: do not spend time and effort merging figures unless you are sure you have nothing else left to do!
% • Font:
% You should use a ‘normal’ font at 11 point or 12 point size. We recommend Helvetica, Arial or Times New Roman – similar fonts are also fine. Do not use highly stylized or bitmap fonts. You may have any number of references, but note that excessive referencing will be as frowned upon as inadequate referencing!
% The main body of the text should use 1.5 line spacing and page numbering should be used.
% The thesis margins should be at least 2 cm and the main text font size should not be smaller than 11 point.








\begin{document}




\vspace{30mm}
\maketitle
\thispagestyle{empty}

\vspace{5mm}
\centerline{Word Count: \wordcount}

\vspace{15mm}
\onehalfspacing
\renewcommand{\abstractname}{\vspace{-\baselineskip}} %hide abstract title

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% Abstract %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% NB: follow the usual thing for this, from nature - best resource I reckon
\begin{abstract}
    \linenumbers
    \noindent
    \textbf{The ability to understand and predict population growth is vital for multiple disciplines. Technology is increasingly enabling us to model large datasets, uncover the insights buried within them, and improve the models iteratively. Scaling up this process to ingest more data and quantify improvements to the models would push forward our capabilities and inform future research. Here I showcase a prototypal pipeline to fit established models to hundreds of datasets, quantify their performance for comparison between them, and use control variable data to glean insights out of this process. Consistent with the literature, the methodology proclaimed the Gompertz model as the highest-performing of those tested, while highlighting its flaws. Correlating performance of the models and separately the morphology of their resultant fits with potential covariables has the potential to improve or even inspire subsequent investigation. Meanwhile the pipeline as a whole can, with modest alterations, be used on groups of models from a multitude of fields, at best facilitating the development of the very models it upon which it is used to analyse and elucidate.
    }
\end{abstract}
\vspace{10mm}


\newpage
\tableofcontents
\thispagestyle{empty}

\newpage
\linenumbers

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% Introduction %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\setcounter{page}{1}
\section{Introduction}
% A good introduction should leave the reader with a clear idea of the problem to be tackled and looking forward to the more detailed sections to follow.
% It should include a section on the general way the problem has been approached. 
% An essential concluding part of the introduction is to clearly define the aims of the research project and any hypotheses tested.
% Also, think about:
%  o What is this paper about? (i.e., the broad area, big picture) Why is that interesting?
%  o Given it’s so interesting, why don’t we know the answer?
%  o So, what is this about, more specifically? What are hypothesised to be the important things? Build from the most general and fundamental hypotheses to the most refined or tenuous ones.
%  o How, roughly and briefly, will you go about testing these hypotheses? Why are you using this system? What approach will you use?
%  o State clearly what your hypotheses are.

History and background of pop genetics (perhaps specifically study of mating patterns)

background of assortative mating and pop structure, significance

sociologogical background of the region, eg SA colonialism, slavery effective, why it's a good case study

assortative mating by social structures eg wealth, status, etc in the region - have claims that it exists be made?

genetic markers telling us about ancestry (assuming that's how RFMix works)



neural networks

use of neural networks on these markers/windows/whatever



Aims of paper






1000ish words should be good, 1500 absolute max



From proposal: 

intro
Certain genetic phenomena, including assortative mating and sex bias, have the potential to alter the structure of human populations.  This in turn modifies genomic variation, reflected in a population’s genomic data which can be used to infer said phenomena.  Various factors,  cultural or socio-economic, can cause them to arise or to manifest themselves.  Historically, the social stratification of societies by wealth, power and perceived race, alongside explicit racial segregation policies, have modulated human mating behaviour away from random mating. In  the  past  century,  the geopolitical  and  economic  landscape  has  experienced  and  will  continue to experience intercontinental migration towards areas of high population density, engendered by the  likes  of  globalization,  industrialisation,  shifting  demographics,  the  fallout  of  colonialism  and global  warming.   This  mass  migration  occurring  within  a  relatively  small  timescale  has  founded new, diverse societies with complex and stratified urban population structures.The  modern  era  is  not  the  first  in  which  this  has  occurred,  indeed  the  convoluted  layers  ofancient processes of migration and subsequent admixture,  which shaped populations around theworld  over  millennia,  are  shown  to  have  been  sex-biased  in  many  cases,  and  may  additionallyhave been impacted by localised assortative mating patterns.(Goldberg et al., 2014; Skoglund and Mathieson, 2018) This project seeks to utilise deep learning algorithms and widely available genomic data in theelucidation of how complex human mating behaviours have been effected by both social and eco-nomic conditions, which stem from the genetic structure of different historic and current admixingpopulations.(Sheehan and Song, 2016)

Outcome
The project’s outcome will be in two forms.  I will be comparing the efficacy of the different neuralnetwork combinations outlined above based on their accuracy estimating multiple parameters.  Iwill then be using highest-performing method to infer parameters, such as assortative mating andsex bias, from genomic data and ultimately integrate my findings into the phenotypic and culturallandscape of the studied region.










% Explain theory behind AMI and window length in introduction!!!


















%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% Methods %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\vspace{5mm}
\section{Methods}

% This should contain details of any methods used extensively during the project, layout of field experiments, theoretical methods, methods of statistical analyses etc.
% You can use subheadings for different procedures or tests.
% If field work is done, a general description of the study area may be included here.
% Extra methodological details can be placed in appendices.
% The golden rule is that the reader should be able to repeat what you did, should they so wish.
% The other rule -- more important for your project than in a paper -- is that you describe in enough detail to show you’ve understood what you did.
% You should feel free to use subheadings in your methods and results to help organise different parts of your project.
% If so, keep the same order of the different parts of the project in all of your sections: the methods for testing each hypothesis and the results of those tests are described in the same order as the hypotheses are described in the introduction.

% Also, think about:
% o What is the overall design of the study?
% o What are the variables and how do they relate to the hypotheses?
% o How did you get the data?
% o What are the characteristics of the data set / experiment -- how many observations, how many replicates etc.
% o General procedures, if any, that are true in all of the analyses (e.g., transformation of data, model checking, how models were compared)
% o How did you test the hypotheses, in the logical order outlined in the introduction (i.e., from the general to the specific)? Make sure you show that your tests are appropriate.

% Computer Programs. If the program has been published, cite the reference, include it in the reference list and provide a brief outline of the methods it uses.
% If you are using a program or code generated for the project then a more complete description is needed in the main text.
% You should provide the code used in an appendix and consider providing a flow chart and usage notes to help interpretation.
% You should take care to define all the input variables used in the program.


\subsection{Studied Populations}


For the initial analyses, all African, European and American populations from the 1000 Genomes Project (1KGP) and the Human Genome Diversity Project (HGDP) were used \textbf{(Table 1)}, with the exception of the Russian and Finnish populations. These were excluded owing to minimal colonial-era migration to the Americas from these populations, alongside the genetic similarities between these populations, Siberans and, by extension, Native Americans.



\begin{table}[htb]
    \centering
    \caption{
        \textbf{Details of the populations used throughout this study.} 
        Populations abbreviated as three capitalised letters are from the 1000 Human Genome Project dataset, while full-word abbreviated populations are from the Human Genome Diversity Project dataset. The number of samples used from each population is denoted by "n". \\
        **The Tuscan and Yoruba populations comprise samples from both datasets.
        }
    \small
    \begin{tabular}{ |p{3cm}||p{8cm}|p{3cm}|p{0.8cm}|  }
    \hline
    \multicolumn{1}{|c||}{\textbf{Superpopulation}} &
    \multicolumn{1}{c|}{\textbf{Population}} & 
    \multicolumn{1}{c|}{\textbf{Abbreviation}} & %vline missing here on purpose
    \multicolumn{1}{c|}{\textbf{n}}\\
    \hline
    \hline
    \multirow{6}{*}{Admixed}  %num == number of cols included
        &African Ancestry in Southwest USA & ASW & 61 \\
        &African Caribbean in Barbados & ACB & 96 \\
        &Colombian in Medellin, Colombia & CLM & 94 \\
        &Mexican Ancestry in Los Angeles, California & MXL & 64 \\
        &Peruvian in Lima, Peru & PEL & 85 \\
        &Puerto Rican in Puerto Rico & PUR & 104 \\
        \hline
    \multirow{11}{*}{African}
        &Bantu in Kenya & BantuKenya & 11 \\
        &Bantu in South Africa & BantuSouthAfrica & 8 \\
        &Biaka in Central African Republic & Biaka & 22 \\
        &Esan in Nigeria & ESN & 99 \\
        &Gambian in Western Division, The Gambia & GWD & 113 \\
        &Luhya in Webuye, Kenya & LWK & 99 \\
        &Mandenka in Senegal & Mandenka & 22 \\
        &Mbuti in Democratic Republic of Congo & Mbuti & 13 \\
        &Mende in Sierra Leone & MSL & 85 \\
        &San in Namibia & San & 6 \\
        &Yoruba in Nigeria & YRI/Yoruba* & 129 \\
    \hline
    \multirow{9}{*}{European}
        &Basque in France & Basque & 23 \\
        &Bergamo Italian in Bergamo, Italy & BergamoItalian & 12 \\
        &British in England and Scotland & GBR & 91 \\
        &Northern and Western European Ancestry in Utah & CEU & 99 \\
        &French in France & French & 28 \\
        &Orcadian in Orkney & Orcadian & 15 \\
        &Sardinian in Italy & Sardinian & 28 \\
        &Iberian in Spain & IBS & 107 \\
        &Toscani in Italy & TSI/Tuscan* & 115 \\
    \hline
    \multirow{5}{*}{Native American}
        &Colombian in Colombia & Colombian & 7 \\
        &Karitiana in Brazil & Karitiana & 12 \\
        &Maya in Mexico & Maya & 21 \\
        &Pima in Mexico & Pima & 13 \\
        &Surui in Brazil & Surui & 8 \\
    \hline
    \end{tabular}
\end{table}



%NB - remember you need to mention all parameters


\subsection{Data Preparation with BCFtools}

Using BCFtools v1.9, the 30x coverage 1KGP and high-coverage HGDP datasets were merged, and all populations except those listed in \textbf{(Table 1)} were removed \parencite{Danecek2021}. All C→G, G→C, A→T and T→A SNPs were filtered out as they are harder to align and are hence prone to error. SNPs were further filtered with a minor allele frequency threshold of 5\%, as to reduce the dataset and remove rare and thus uninformative SNPs. Following this, all 22 filtered VCF files, one per autosome, were indexed for phasing. 








\subsection{Haplotype Estimation with SHAPEIT4}


Phasing was carried out using SHAPEIT4.2.0, which efficiently assigns haplotype estimates for each genotype by cross-referencing the genomic region in question with the corresponding region of a pre-phased reference panel and of the other genomes being phased \parencite{Delaneau2019}. The programme was run using the B38 genetic map recommended by the developers and default parameters, plus an apropriate high-coverage phased reference genome from the 1KGP website (see: \textbf{Data and Code Availability}) to inmprove haplotype estimation accuracy. The individually phased chromosomes were then merged into a single VCF file with BCFtools. 





\subsection{Ancestry Estimation with PLINK \& ADMIXTURE}


Linkage disequilibrium pruning was performed with PLINK on the genomes in VCF format, which creates a subset of more or less independent SNPs - thereby significantly reducing the computational power needed for subsequent analyses with minimal information loss - before converting the pruned dataset to PLINK format \parencite{Purcell2007}.
The programme ADMIXTURE v1.3.0 used cluster analysis and principal component analysis to estimate the proportions of African, European and Native American ancestry for each remaining sample, with default parameters and three ancestries to be detected \parencite{Alexander2009}.





\subsection{Local Ancestry Inference with RFMIX v2}


The ADMIXTURE outputs were subsequently used to filter out all significantly admixed samples, with a minimum threshold of 99\% African, European or Native American ancestry. This subsetting was executed using BCFTools, yielding a subset VCF of >99\% pure samples was used as a reference panel for local ancestry assignment with the programme RFMIX. A query subset was created correspondingly, containing all samples in the "Admixed" superpopulation in \textbf{(Table 1)}.

RFMIX v2.03-r0, based on concepts developed in RFMIX v1, assigns ancestries to segments of an individual's genome, which not only yields ancestry proportions as with ADMIXTURE, but also effectively maps out each genome in terms of each genomic region's estimated ancestry or origin \parencite{Maples2013}. It does this by subjecting the chromosomes to a combination of machine learning methods: discriminant random forests and conditional random field modelling.

The RFMIX run was performed using the aforementioned query and reference VCF files, and a sample map linking the sample codes to their respective populations. Parameters used were 20 generations, before which no known European-Native American admixtire had taken place, and three runs of the algorithm.





\subsection{Assortative Mating Index Calculation}


One measure of assortative mating is the assortative mating index (AMI), which takes a log odds ratio of the relative local ancestry homozygosity and heterozygosity:


\begin{equation}
    AMI = ln{\left( \frac{ hom^{\: obs} / hom^{\: exp} }
                         { het^{\: obs} / het^{\: exp} } \right)}
\end{equation}
\vspace{3mm}


Three ancestries are being investigated, hence expected homozygous and heterozygous allelic frequencies can be thought of in terms of the biallelic (\textbf{(Equation ???)}) or triallelic (\textbf{(Equation ???)}) Hardy-Weinberg models \parencite{Norris2019}: 


\begin{equation}
    (a + e + n)^{2} = a^{2} + e^{2} + n^{2} + 2ae + 2an + 2en
\end{equation}


\begin{equation}
    (x + y)^{2} = x^{2} + 2xy + y^{2}
\end{equation}
\vspace{3mm}


Here a, e and n in the triallelic model are initials of the ancestries they signify, while x and y in the biallelic model correspond to a given ancestry - African, European or Native - and all other ancestries. Hence, while AMI is calculated only one time using the triallelic model, the AMI using the biallelic model must be calculated three times: once with respect to each ancestry. For example, with respect to African ancestry, the homozygous genotype would be both African alleles or both non-African alleles, and the heterozygous genotype would be one African allele and one of the other Ancestry's alleles.

The outputs of RFMIX were analysed by a series of R Studio scripts I created for this project (see: \textbf{Data and Code Availability}). Firstly, the forward-backward (.fb.tsv) ouput files were read by the script "rfmix.fb.tsv_genotype_assign_HPC.R". These files contain the estimated haplotype probabilities at each genolmic position for each sample. The script then assigns the genotype for each genomic position in each sample, with a probability threshold of 0.9, and returns the frequencies of each of the six triallelic genotypes at each position across samples as a table. This genotype frequency table is then read by the script "rfmix.fb.tsv_genotype_analysis.R", before calculating the triallelic AMI, and the three biallelic AMIs with respect to each ancestry, at each position.






\subsection{Genomic Fragment Length Analysis}

Another indicator of assortative mating is analysis of ancestry tract lengths: the length of genomic fragments constituting consecutive assignments of the same ancestry. The distribution of these lengths can hint at how long ago and to what extent admixture occured which, when compared with migration data, can indicate if and when assortative mating occured \parencite{Gravel2012}
.

Assignments of lower certainty in the forward-backward file, using the 0.9 probability threshold, have the potential to fracture fragment lengths and completely alter the fragment length distribution. Hence the .msp.tsv RFMIX output files were used instead, equivalent to the forward-backward files but with automatic haplotype assignment to haploytype with highest estimated liklihood.

To do generate the fragment length distributions, the script "rfmix.msp.tsv_window_size.R" read the .msp.tsv files, summed the length of consecutive genomic windows assigned to the same ancestry, and appends the lengths to the vector containing the lengths of other fragments corresponding to the fragment's ancestry and population. The script "rfmix.msp.tsv_inbred_window_size.R" works similarly, but generates fragment length distributions of consecutive homozygous genotype assignments, rather than haplotype assigmnets. 





\subsection{Timeline of Admixture Estimation with TRACTS}

TRACTS is a software for modelling migration histories using ancestry tracts data, incorporating the theory of time-dependent gene-flow and correcting for chromosomal end effects and haplotype assignment errors \parencite{Gravel2012}. 

The software uses the .bed file format as input, a file output of the original RFMIX but not of RFMIX v2. 
%explain briefly how the conversion script works, and what .bed files entail. also mention the voyage data, and how the generation data was calculated.


\parencite{Gravel2012}








%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% Results %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Results}

% Describe your results in a logical order: this may not necessarily be the order in which you did the experiments.
% Briefly summarise the main results at the end of each main experiment or sequence of associated experiments.
% Do not duplicate results -- put a table or a graph but not both unless the two methods of presentation demonstrate different points of importance.
% You must refer appropriately to figures or tables in the text and remember to emphasise and perhaps quote significant results.

% In particular, think about:
% o What were the results of your hypothesis tests, in the order you describe them in the Methods?




% Figures:
% You should prepare figures to the same standard required for publication.
% All journals provide advice on preparing figures for publication, so do look at the advice to authors pages for your chosen journal.
% All figures must be numbered and have a caption that is sufficiently detailed to explain the main features of the content by itself.
% All figures must be referred to in the main text of the thesis.
% Put the figures in appropriate points in the text, close to the text that refers to them.

% In particular:
% o The resolution of your figures is crucial. For plots, try to use vector image formats (exported as svg, pdf, or eps) and not bitmapped (raster) formats like JPG and TIFF. Standard /LaTex documents typically allow *.eps or *.pdf figures to be inserted. Using the freely available (and very capable!) vector graphics program Inkscape to ``fine-tune'' your figures is often a good idea.
% Inkscape will also allow svgs to be exported in a /LaTex compatible format (see the Inkscape documentation). For RASTER graphics, the freely available GIMP editor works very well.
% o When using Word, figures in Windows Metafile format are the most reliable vector format. For Word 2011 on Mac, figures in PDF format should give a good result. If you do have to use bitmaps, make sure they are at a high resolution (300 dpi or more) -- this can be particularly important if you need to present line drawings or photographs of specimens or equipment.
% o Plots are all about the data, so reduce margins and maximise the space in the figure for showing the data.
% o Create the figure at the right size -- when it is included in your thesis are all the axis labels and text going to be clearly legible.
% o Avoid `chartjunk' (google Edward Tufte!) -- and avoid superfluous lines, legends and titles along with 3D effects.

% Tables:
% Each table should be numbered, have a full descriptive caption and again must be referred to in the main text.
% Column headings should state units of measurement.
% Avoid large, complicated tables in the main thesis and if you have a large body of numerical data put it in an appendix.



%Example Figure

% \begin{figure}[htb!]
%     \centering
%     \includegraphics[width=\textwidth]{../results/figures/8plots.pdf} 
%     \caption{\textbf{Exemplary timeseries plotted with the regression line that each model fit to them.}  Plots showing log\textsubscript{2} of the population measurements against time in hours, which demonstrate to what extent each model can tolerate peculiar datsets. Linear models fit datsets comprising very few datapoints with near-perfect accuracy, yet without describing the true relationship at all (top left). Timeseries with a death phase and/or no lag phase are modelled poorly by the NLMs which plateau prematurely, although the logistic model benefits from a lack of lag phase (top right). For timeseries with lag phases but which had not plateaued when measurement ceased due to a drawn-out transition between exponential and stationary phases, logistic fails to capture the lag phase and Buchanan simply plateaus at the final datapoint, while Baranyi and Gompertz plateau harshly at the start of the transition, the latter to a lesser extent (bottom).}
% \end{figure}




Check methods for any results, eg numbers of samples filtered at various points 

ADMIXTURE: (African, European or Native American ancestry. This subset of  1129 samples - 550, 507 and 72 samples respectively used for RFMIX)



Overall AMI plot is trallelic and sums all the hets and all the homs, while the ancestry plot is biallelic with respect to one ancestry each time.

















%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% Discussion %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Discussion}

% This should attempt to tie together the results, what they indicate in a broader context, the extent to which the original aims have been satisfied and what future work is suggested.
% Return to and address the ideas raised in the introduction.

% In particular, think about:
% o What’s the main thing we know now that we didn’t know before?
% o What’s the chain of logic and results that means we know it?
% o How does this affect our -- and other scientists’ -- view of the world? What are the implications?
% o What are the implications of the intermediate steps in the chain towards the main thing?
% o What are the caveats that apply to this study? (Leave out caveats that apply to all studies.) What might be done about them? (Very important in a project write-up -- What would you do differently if you were doing the project again or had more time?)
% o What future work could build more broadly on what we’ve found?
% o A nice wrap-up, emphasising how this study in this system is of interest to people who work on other things, or other systems.










%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%% NB DO NOT DELETE!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
stuff for discussion (main project)
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
QUESTION
% hap1_afr=1      hap1_eur=0      hap1_nat=0      hap2_afr=1     hap2_eur=0     hap2_nat=0  

In example above, is it not possible the two african haplotypes are different alleles? Which would technically make it heterozygous without us knowing?
(see email 20th may for more context)

ANSWER
If you are analyzing the nucletotides, you don't know. It could be a fragment from African ancestry and be a T and a fragment from European ancestry and be a T as well. You have to look to the vcf. This applies to your second example also.
It has the problem that unadmixed admixture sources from sub-Saharan Africa have higher effective population size and therefore higher heterozygosity of nucleotides. So there will be a correlation of higher sub-Saharan ancestry and higher heterozygosity of nucleotides, and therefore a bias. However, it might be interesting to discuss how heterozygosity only is not a good indicator of assortative mating in an admixed population. 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
Remember above 99pc PEL individuals are included in ref sample, hence assignment of PEL ancestry will be a bit weird - definitely a bias. I went forward with both populations, but the one with will match native very perfectly, and the one without will A. show as less native than it truly is as a population as v native ones are excluded, and B. might have a native bias in assignment, where there are similarities between two PEL individuals which may not be a general native trait but a specific PEL trait. 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

For frag length histograms, 2nd peaks suggest another migration event, as one would expect it to simply be a normal distribution if only one migration event occured. Similarly, right-tailed distributions suggest constant strean of subsequent immigrants after main migration event (and vice versa). test.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

Remember, ASW is Americans of Sub-Saharan African Ancestry in Oklahoma, Southwest USA, MXL is Mexican Ancestry in Los Angeles CA United States - so significant sample bias; ASW will have more african than the average US SW resident; MXL will have more European, and possibly african could have come later when in LA vs generations ago in Mexico

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

More NAT samples would be better; 72 vs 507 Eur and 550 Afr


















%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% Data/Code %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


\section{Data and Code Availability}

% name a data and a code (GitHub) archive from where the data and code can be obtained that will allow replication of your results. The code may be in the form of a single script file. You will be taught the principles of reproducible analyses in the R week of your coursework. If the data cannot be made available publicly (e.g., because it is yet to be formally published), or if there are some other confidentiality issues with submitting the data, speak with your course director and supervisor, and include a clear statement about why the data cannot be made available under the same Code and Data Availability header.
\subsection{Data}

1KGP Samples: https://www.internationalgenome.org/data-portal/data-collection/30x-grch38
HGDP Samples: https://www.internationalgenome.org/data-portal/data-collection/hgdp
Phasing Reference Panel: http://ftp.1000genomes.ebi.ac.uk/vol1/ftp/data_collections/1000G_2504_high_coverage/working/20201028_3202_phased/
Phasing Genetic Map: https://github.com/odelaneau/shapeit4/blob/master/maps/genetic_maps.b38.tar.gz

\subsection{Code}

Code: https://github.com/Bennouhan/cmeecoursework/tree/master/project/code




\newpage
\printbibliography[heading=bibintoc]
\newpage














%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% Sup Material %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section*{Supplementary Material} % * prevents numbering
\addcontentsline{toc}{section}{Supplementary Material} %add to table of contents

% You may provide Supplementary Information (SI) to provide parts of the study not directly relevant to the main narrative: detailed methods, mathematical derivations, details of computer algorithms, long tables of detailed results, and taxonomic descriptions, lists and drawings in an otherwise ecological study.
% For example, a molecular study might state in the Methods section of the main text that you extracted DNA according to a phenol/chloroform extraction protocol according to a particular reference.
% In the SI, you should then describe the steps of your lab protocol in sufficient detail that other people could reproduce this procedure by following your description.
% Similarly, you should put long tables of results in the main text (these should be in SI); only summary tables or graphs and key results of analysis should appear in the main text.
% However, the project markers are not obliged to read the SI, so the text in the main manuscript should detail everything that the marker needs to know.
% The SI should be presented as an additional document and must be concatenated to the end of the main thesis pdf file before submission (that is, a single pdf file must be submitted).
% Make sure that the SI is neatly formatted (using the same style as the main text), and that all Sections, Tables and/or Figures of the SI are appropriately cited in the main text.


% Computer Programs: If the program has been published, cite the reference, include it in the reference list and provide a brief outline of the methods it uses.
% If you are using a program or code generated for the project then a more complete description is needed in the main text.
% You should provide the code used in an appendix and consider providing a flow chart and usage notes to help interpretation.
% You should take care to define all the input variables used in the program.


\end{document}